core/src/main/java/com/linecorp/armeria/server/HttpStatusException.java

```diff
@@ -17,14 +17,46 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.util.function.Function;
+
 import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.annotation.Nullable;
 
 /**
  * A {@link RuntimeException} that is raised to send a simplistic HTTP response with minimal content
  * by a {@link Service}. It is a general exception raised by a failed request or a reset stream.
  *
+ * <p>Note that an {@link HttpStatusException} raised may not be applied to the next decorators if the
+ * {@link HttpStatusException} is not recovered before passed to the next decorator chain.
+ * For that reason, you need to properly handle the thrown {@link HttpStatus} into a normal
```

Comment:
since `raised` is describing the exception, `HttpStatus` -> `HttpStatusException`, better not to use pronouns "you", "I" in academic writing

```suggestion
 * <p>Note that a raised {@link HttpStatusException} may not be applied to the next decorators if the
 * {@link HttpStatusException} is not recovered before being passed to the next decorator chain.
 * For this reason, the thrown {@link HttpStatusException} should be converted into a normal
```

Reply:
TIL. ❤️